### ISSUE TITLE
Ignore.If

### ISSUE THREAD

**Issue by NUnitTester** [nunit/nunit#4369]
There is the possibility to use
`Warn.If(condition, message)`

Often I want to Ignore a Test depends on some environment issues (version of database, system, ...)
So I also want to write:
`Ignore.If(condition, message)`

Up to now I have to use an if-Statement and write `Assert.Ignore(message)`

**Comment by stevenaw** [comment]
At first glance this seems like an inconsistency in the API and I can understand why you would ask us to change it.

Underneath, however, `Warn` and `Ignore` are actually doing quite different things. `Warn` is handled as an _assertion_ which is tracked as part of the tests' execution and expected to vary throughout the test. `Ignore` on the other hand, is a higher-level property of the test method itself and based on the summary within https://github.com/nunit/nunit/issues/3118#issuecomment-449568209 is that "Adding a property to a test during the test was never intended to be exposed functionality". I think this is why the guidance in the docs for [`Assert.Ignore()`](https://docs.nunit.org/articles/nunit/writing-tests/assertions/classic-assertions/Assert.Ignore.html) is that "We recommend that you use this only in isolated cases.".

Can you explain a bit more under what conditions you would want to dynamically ignore your tests? I'm curious if perhaps using a `TestCase` or `TestCaseSource` and dynamically setting the `Ignore` property there could also work for you

**Comment by OsirisTerje** [comment]
@NUnitTester 
You mean you want something like this:  
`
public class TestsIgnore
{
    [TestCase(41)]
    [TestCase(42)]
    public void Test3(int x)
    {
        Ignore.If(x == 42, "This is a warning");
        TestContext.WriteLine($"Was not ignored {x}");
    }
}


public static class Ignore
{
    public static void If(bool condition, string message)
    {
        if (condition)
        {
            Assert.Ignore(message);
        }
    }
}

`
Resulting in something like:
![image](https://user-images.githubusercontent.com/203432/235233278-f9386bb5-83d1-4453-a2b1-a970fd0e3829.png)

**Comment by NUnitTester** [comment]
> Can you explain a bit more under what conditions you would want to dynamically ignore your tests? I'm curious if perhaps using a `TestCase` or `TestCaseSource` and dynamically setting the `Ignore` property there could also work for you

I have mentioned that we used it for differences in environment. Some Tests which are testing localizations only work if the system has german, others need english system. The other environment part could be the database. Some rare tests are specific for oracle or sql server, some are specific for a special version. Some tests are specific for the computer because a developer computer has no deploy build and the biuld server has.

**Comment by NUnitTester** [comment]
> @NUnitTester You mean you want something like this:

@OsirisTerje Thanks! That's what I want to have.

I know that it is possible for testCases like this:
`yield return new TestCaseData(...).Ignore("test to be ignored")`

**Comment by OsirisTerje** [comment]
Yes, and you could add that to your own test extensions library.  That said, it would be cool to have a NUnit.Extensions library to pack these kinds of functions, a kind of tool library with all sorts of goodies.  There are no such yet though.
The framework itself should not have too many of these but concentrate on the fundamentals.  That makes maintenance easier.
